packages/node_modules/@ciscospark/react-component-people-list/src/index.js

```diff
@@ -16,56 +19,99 @@ const propTypes = {
       isPending: PropTypes.bool
     }))
   })).isRequired,
-  onItemClick: PropTypes.func
+  onItemClick: PropTypes.func,
+  onDisplayUsers: PropTypes.func
 };
 
 const defaultProps = {
-  onItemClick: () => {}
+  onItemClick: () => {},
+  onDisplayUsers: () => {}
 };
 
-
 function PeopleList({
   items,
-  onItemClick
+  onItemClick,
+  onDisplayUsers
 }) {
   if (!items || !items.length) {
     return null;
   }
 
-  function generateGroupList(people) {
-    return people.map((p) => {
+  const totalRows = [];
+
+  items.forEach((item) => {
+    // Header for item
+    if (item.label) {
+      totalRows.push({type: 'header', details: item.label});
+    }
+    // Rows for each person
+    item.people.forEach((person) => {
+      totalRows.push({type: 'person', details: person});
+    });
+  });
+
+  function rowRenderer(options) {
+    const {
+      key,
+      index,
+      style
+    } = options; {
+      if (totalRows[index].type === 'header') {
+        return (
+          <div style={style} className={classNames('ciscospark-people-group', styles.group)} key={totalRows[index].details}>
+            {
+              totalRows[index].details
+              && <h3 className={classNames('ciscospark-people-group-title', styles.title)}>{totalRows[index].details}</h3>
+            }
+          </div>
+        );
+      }
+
+      const person = totalRows[index].details;
+
       const handleItemClick = () => {
-        onItemClick(p);
+        onItemClick(person);
       };
 
       return (
-        <Person
-          displayName={p.displayName || p.name}
-          key={p.id}
-          onClick={handleItemClick}
-          {...p}
-        />
+        <div key={key} style={style}>
+          <Person
+            displayName={person.displayName || person.name}
+            key={person.id}
+            onClick={handleItemClick}
+            {...person}
+          />
+        </div>
       );
-    });
+    }
+  }
+
+  function onRowsRendered({startIndex, stopIndex}) {
+    const userIds = totalRows.slice(startIndex, stopIndex).filter((row) => row.type === 'person').map((p) => p.details.id);
```

Comment:
Could you throw a quick comment on what this is doing?

Reply:
yepp, will do